dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java

```diff
@@ -438,4 +394,20 @@ public Map<Long, String> queryWorkerGroupByWorkflowDefinitionCodes(List<Long> wo
                 Schedule::getWorkerGroup));
     }
 
+    private void boardCastToMasterThatWorkerGroupChanged() {
+        final List<Server> masters = registryClient.getServerList(RegistryNodeType.MASTER);
+        if (CollectionUtils.isEmpty(masters)) {
+            return;
+        }
+        for (Server master : masters) {
+            try {
+                Clients.withService(IMasterContainerService.class)
+                        .withHost(master.getHost() + ":" + master.getPort())
+                        .refreshWorkerGroup();
+            } catch (Exception e) {
+                log.error("Broadcast to master: {} that worker group changed failed", master, e);
```

Comment:
If there are occasional network interruptions between the API server and master, should retries be implemented during broadcasting? Otherwise, master will take 5 minutes to update.

Reply:
> If there are occasional network interruptions between the API server and master, should retries be implemented during broadcasting? Otherwise, master will take 5 minutes to update.

I think if users needs a faster update just adjust 5 minutes to a smaller value, which allows user-defined.

Reply:
The max delay is 5 minutes, user can reduce this by modified the value at master's config. I don't want to add retry here, since in most cases, the network should work well, otherwise we may need to add retry at rpc framework.

Reply:
> add unify retry at rpc framework

LGTM.  Maybe we can create a new issue about this? @ruanwenjun

Reply:
I create #16870 to track this.